src/cdk/text-field/autosize.ts

```diff
@@ -203,6 +211,13 @@ export class CdkTextareaAutosize implements AfterViewInit, DoCheck, OnDestroy {
     textarea.placeholder = placeholderText;
 
     this._previousValue = value;
+
+    if ((textarea.scrollHeight !== this._previousHeight && !this._maxHeight)
```

Comment:
Can you pull out all of these references to `textarea.scrollHeight` into a variable? Requesting it will cause the styles to be recalculated synchronously. See https://gist.github.com/paulirish/5d52fb081b3570c81e3a